internal/app/networkd/pkg/address/kernel.go

```diff
@@ -0,0 +1,181 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package address
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/talos-systems/talos/internal/pkg/kernel"
+)
+
+// Kernel implements the Addressing interface
+type Kernel struct {
+	Addr              string
+	ServerAddress     string
+	Gateway           string
+	Netmask           string
+	FQDN              string
+	Device            string
+	Auto              string
+	PrimaryResolver   string
+	SecondaryResolver string
+	NTPServer         string
+}
+
+// Discover doesnt do anything in the static configuration since all
+// the necessary configuration data is supplied via config.
+// nolint: gocyclo
+func (k *Kernel) Discover(ctx context.Context) error {
+	var (
+		option *string
+		err    error
+	)
+
+	if option = kernel.ProcCmdline().Get("ip").First(); option == nil {
+		return fmt.Errorf("%s", "no kernel.ip argument supplied, skipping")
+	}
+
+	// TODO may need to move this earlier in networkd; if ip.dhcp is specified
+	// we shouldnt do anything because we dont know which (all?) interfaces the kernel
+	// will autoconfigure
+	if *option == "dhcp" {
```

Comment:
If the option is DHCP, then we really should just fall back to our default behaviour, which is DHCP anyway.  I don't think that constitutes an error.

Reply:
+1